src/buffer/cylinder-buffer.ts

```diff
@@ -49,6 +49,12 @@ class CylinderBuffer {
   }
 }
 
+const CylinderBuffer: {
+  new(data: CylinderBufferData, params: Partial<CylinderBufferParameters>): CylinderGeometryBuffer | CylinderImpostorBuffer;
+} = CylinderBufferImpl as any;
+
+type CylinderBuffer = CylinderGeometryBuffer | CylinderImpostorBuffer;
```

Comment:
In my tests, this type declaration did not seem necessary. Also, it might cause some confusion with the class wrapper name.

Reply:
`CylinderBuffer` is used as a type in a number of places - for example, `UnitcellRepresentation#cylinderBuffer` is typed as `CylinderBuffer`. If you prefer, I can change all instances of the CylinderBuffer type to be `CylinderGeometryBuffer | CylinderImpostorBuffer`, but this just seemed to me to be a handy shortcut (especially as the idea behind this is that you think of it as a CylinderBuffer in general, and don't care whether it's a GeometryBuffer or an ImpostorBuffer)

Reply:
@ppillot Any update on this?

Reply:
Sorry!
I am still feeling queasy about this. I can see that the PR works. It is unclear to me though how the 2 symbols, one for typings, the other for a constructor, are coexisting. There is some documentation in typescript reference [about declaration merging](https://www.typescriptlang.org/docs/handbook/declaration-merging.html) but the specific case presented here is not specified (merging const name and type name). I can't tell if this is guaranteed to work in the future.
Overall I am indecisive regarding this PR. I won't object for it to be merged. @garyo @fredludlow any opinions?

Reply:
`const CylinderBuffer` is not a type, it is a runtime value. Similarly `type CylinderBuffer` is a type, not a runtime value. AFAIK, you are allowed to define type and runtime symbols with different names, and a type is only implicitly created for a runtime value in the case of a class, enum, etc.

I get your concern though - as I said, if the preference is to change all `CylinderBuffer`s to `CylinderGeometryBuffer | CylinderImpostorBuffer`, we can do that.

Reply:
Oh I see what you're saying wrt those docs, I misread your comment. Yeah my understanding is that TS treats type names and value names as separate sets of symbols, and creates them according to the rules given in the table.

Reply:
Let me know whether the desired outcome is preserving `CylinderBuffer` or changing usages of that `CylinderGeometryBuffer | CylinderImpostorBuffer`, and I can make any changes as needed